src/middlewared/middlewared/plugins/network.py

```diff
@@ -1706,8 +1706,76 @@ def choices(self, options):
         except Exception:
             self.logger.info('Failed to sync routes', exc_info=True)
 
+        if not await self.middleware.call('system.is_freenas') and await self.middleware.call('failover.licensed'):
+            await self.nic_capabilities_check()
+
         await self.middleware.call_hook('interface.post_sync')
 
+    @private
+    async def nic_capabilities_check(self):
```

Comment:
A docstring explaining why this is all needed might com in handy if someone else needs to poke at tihs in the future

Reply:
Yes, also please name all methods accordingly so they can be reused in the future. `nic_capabilities_check` - not very clear, check for what? `check_nic_capabilities_for_xxx` will be much better. Or even `disable_evil_nic_capabilities` (because you call `disable_capabilities` at the bottom).